crates/fuel-core/src/state/historical_rocksdb.rs

```diff
@@ -197,6 +202,20 @@ where
         storage_transaction: &mut StorageTransaction<&RocksDb<Historical<Description>>>,
         height: &Description::Height,
     ) -> StorageResult<()> {
+        let lock_guard = self.migration_in_progress.read();
```

Comment:
Probably not a big deal, but we most likely don't need to keep this locked for the entire function. Otoh, it should do no harm.